File: python/consumer.py
import configparser
import json
import six
import os
from google.cloud import language
from google.cloud.language import enums
from kafka import KafkaConsumer

config = configparser.RawConfigParser()
config.read('../config.cfg')

KAFKA_TOPIC = config.get('Kafka', 'topic')
GOOGLE_APPLICATION_CREDENTIALS = config.get(
    'GCP', 'google_application_credentials')

os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = GOOGLE_APPLICATION_CREDENTIALS


def main():
    consumer = KafkaConsumer(KAFKA_TOPIC)
    client = language.LanguageServiceClient()

    for msg in consumer:
        output = []
        output.append(json.loads(msg.value))

        if 'text' in output[0]:
            content = output[0]['text']

            if isinstance(content, six.binary_type):
                content = content.decode('utf-8')

            type_ = enums.Document.Type.PLAIN_TEXT
            document = {'type': type_, 'content': content}

            response = client.analyze_sentiment(document)
            sentiment = response.document_sentiment

            print(f'Text: {content}')
            print(f'Score: {sentiment.score}')
            print(f'Magnitude: {sentiment.magnitude}\n')


if __name__ == "__main__":
    main()

File: nodejs/consumer.js
const ConfigParser = require('configparser');
const language = require('@google-cloud/language');
const { SimpleConsumer } = require('no-kafka');
const fs = require('fs');

const config = new ConfigParser();
config.read('../config.cfg');

KAFKA_TOPIC = config.get('Kafka', 'topic');
KAFKA_ENDPOINT_PORT = config.get('Kafka', 'kafka_endpoint_port');
GOOGLE_APPLICATION_CREDENTIALS = config.get(
  'GCP',
  'google_application_credentials'
);

process.env['GOOGLE_APPLICATION_CREDENTIALS'] = GOOGLE_APPLICATION_CREDENTIALS;

const clientGoogle = new language.LanguageServiceClient();
const dir = './records';

function initKafka() {
  createFolder();

  const consumer = new SimpleConsumer({
    connectionString: `localhost:${KAFKA_ENDPOINT_PORT}`
  });

  const data = messageKafka => {
    messageKafka.forEach(async ({ message }) => {
      const value = message.value.toString('utf8');
      const { text } = JSON.parse(value);

      if (text) await analyzeText(text)
    });
  };

  return consumer.init().then(() => {
    return consumer.subscribe(KAFKA_TOPIC, data);
  });
}

async function analyzeText(text) {
  const document = {
    content: text,
    type: 'PLAIN_TEXT'
  };

  const [result] = await clientGoogle.analyzeSentiment({ document });
  const { score, magnitude } = result.documentSentiment;
  const data = JSON.stringify({text, score, magnitude})

  createFile(data)
}

function createFolder() {
  if (!fs.existsSync(dir)){
    fs.mkdirSync(dir);
  }
}

function createFile(data) {
  fs.writeFile(`${dir}/twitter-${Date.now()}.txt`, data, (err) => {
    if (err) console.log(err);
    console.log("Successfully written to file.");
  });
}

initKafka();
